addon/components/service-worker-update-notify.js

```diff
@@ -1,94 +1,22 @@
-import Ember from 'ember';
-import { getOwner } from '@ember/application';
 import Component from '@ember/component';
-import { computed } from '@ember/object';
-import { task, timeout } from 'ember-concurrency';
-
 import layout from '../templates/components/service-worker-update-notify';
-import serviceWorkerHasUpdate from '../utils/service-worker-has-update';
-
-const configKey = 'ember-service-worker-update-notify';
-
-async function update() {
-  if (Ember.testing) {
-    return;
-  }
-
-  const reg = await navigator.serviceWorker.register(
-    '{{ROOT_URL}}{{SERVICE_WORKER_FILENAME}}',
-    { scope: '{{ROOT_URL}}' }
-  );
-
-  reg.update();
-}
+import { inject as service } from '@ember/service';
 
 export default Component.extend({
   layout,
   tagName: '',
+  serviceWorkerUpdate: service(),
```

Comment:
should this service be namespaced to both prevent potential collisions, and to let people know where the service comes from?
```ts
updateService: service('service-worker-update-notify/update-service'),
```

Reply:
I would expect `service-worker-update` to be quite unique!? Maybe `service-worker-update-notify` to match the addon name? 

Fwiw, addons can run into this problem always, with a lot of examples where collisions are much more likely than here (e.g. `session` service from `ember-simple-auth`).

Reply:
Renamed to `service-worker-update-notify`